core/server/api_container/server/startosis_engine/interpretation_time_value_store/service_value_repository.go

```diff
@@ -0,0 +1,113 @@
+package interpretation_time_value_store
+
+import (
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/database_accessors/enclave_db"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_types"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	serviceInterpretationValueBucketName = []byte("service-interpretation-value")
+	emptyValue                           = []byte{}
+)
+
+type serviceInterpretationValueRepository struct {
+	enclaveDb          *enclave_db.EnclaveDB
+	starlarkValueSerde *kurtosis_types.StarlarkValueSerde
+}
+
+func getOrCreateNewServiceInterpretationTimeValueRepository(
+	enclaveDb *enclave_db.EnclaveDB,
+	starlarkValueSerde *kurtosis_types.StarlarkValueSerde,
+) (*serviceInterpretationValueRepository, error) {
+	if err := enclaveDb.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucketIfNotExists(serviceInterpretationValueBucketName)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred while creating the bucket for the service interpretation time value repository")
+		}
+		logrus.Debugf("Recipe result bucket: '%+v'", bucket)
+
+		return nil
+	}); err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while building service interpretation time value repository")
+	}
+
+	repository := &serviceInterpretationValueRepository{
+		enclaveDb:          enclaveDb,
+		starlarkValueSerde: starlarkValueSerde,
+	}
+
+	return repository, nil
+}
+
+func (repository *serviceInterpretationValueRepository) AddService(name service.ServiceName, service *kurtosis_types.Service) error {
```

Comment:
nit: suggest renaming `AddService` to `PutService` bc `AddService` means something else in the startosis engine (add in service network) so it's an overloaded concept